tests/unit/pipeline/test_ner_re_pipeline.py

```diff
@@ -0,0 +1,390 @@
+
+    original_relations = document_with_relations["binary_relations"]
+    assert len(original_relations) == 2
+    original_relations_predictions = document_with_relations["binary_relations"].predictions
+    assert len(original_relations_predictions) == 1
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document], other_docs=[document_with_relations], layer_names=["binary_relations"]
+    )
+
+    prepared_doc = prepared_docs[0]
+
+    relations = prepared_doc["binary_relations"]
+    assert len(relations) == 2
+
+    # from predictions
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        from_predictions=True,
+    )
+
+    prepared_doc = prepared_docs[0]
+    relations = prepared_doc["binary_relations"]
+    assert len(relations) == 1
+
+    # to predictions
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        to_predictions=True,
+    )
+
+    prepared_doc = prepared_docs[0]
+    relations = prepared_doc["binary_relations"].predictions
+    assert len(relations) == 3  # adds to previously existing predictions
+
+    # inplace
+
+    prepared_docs = add_annotations_from_other_documents(
+        docs=[document],
+        other_docs=[document_with_relations],
+        layer_names=["binary_relations"],
+        inplace=True,
+    )
+
+    relations = document["binary_relations"]
+    assert len(relations) == 2
+
+
+@dataclasses.dataclass
+class TextDocumentWithEntitiesAndRelations(TextBasedDocument):
+    entities: AnnotationLayer[LabeledSpan] = annotation_field(target="text")
+    relations: AnnotationLayer[BinaryRelation] = annotation_field(target="entities")
+
+
+@pytest.fixture
+def old_document():
+    document = TextDocumentWithEntitiesAndRelations(
+        text="Harry lives in Berlin, Germany. He works at DFKI.", id="tmp"
+    )
+
+    document.entities.extend(
+        [
+            LabeledSpan(start=0, end=5, label="PERSON"),
+            LabeledSpan(start=15, end=30, label="LOCATION"),
+            LabeledSpan(start=44, end=48, label="ORGANIZATION"),
+        ]
+    )
+    assert str(document.entities[0]) == "Harry"
+    assert str(document.entities[1]) == "Berlin, Germany"
+    assert str(document.entities[2]) == "DFKI"
+
+    document.relations.extend(
+        [
+            BinaryRelation(
+                head=document.entities[0],
+                tail=document.entities[1],
+                label="lives_in",
+            ),
+            BinaryRelation(
+                head=document.entities[0],
+                tail=document.entities[2],
+                label="works_at",
+            ),
+        ]
+    )
+
+    return document
+
+
+@pytest.mark.slow
+def test_ner_re_pipeline(old_document):
```

Comment:
We have `inplace` in all methods, maybe we just have it in inside `__call__` method of `ner_re_pipeline` and remove from all other methods ?

Reply:
Hmm I recommend to have it everywhere where it impacts the logic in any way. So also try with and without inplace in the individual test methods. I think you can parametrize them in the same way?

Reply:
So when I pass `inplace` lets say as **True** to the `__call__` method, it does not create new copies of documents. But when these documents are passed to the `process_pipeline_steps` method, default value for `inplace` is **False**, so new documents are created. So maybe we remove `inplace` from `__call__` method because it is ineffective.

Reply:
Yeah, simplifying this sounds valid. But I would propose to remove the parameter for all step related, methods, they should just work **inplace** because this is less work. Also, they should not return anything (this could cause confusion for the user if there are "two ways" to access the result), except for `process_pipeline_steps`. However, the `__call__` should still have the inplace parameter because this follows the implementation of the [default prediction pipeline](https://github.com/ArneBinder/pytorch-ie/blob/main/src/pytorch_ie/pipeline.py ).

Reply:
I added some more simplifications, see [03383ba](https://github.com/ArneBinder/pytorch-ie-hydra-template-1/pull/155/commits/03383ba7e153955703309439bf6250a09a56c8e0).